internal/pkg/parser/virtualservice.go

```diff
@@ -24,22 +25,29 @@ func parseVirtualServices(files []string) ([]*v1alpha3.VirtualService, error) {
 		// we need to transform yaml to json so the marshaler from istio works
 		jsonBytes, err := yaml.YAMLToJSON(fileContent)
 		if err != nil {
-			log.Debug("error converting yaml to json", zapcore.Field{Key: "file", String: file})
+			log.Debug("error converting yaml to json", zapcore.Field{Key: "file", Type: zapcore.StringType, String: file})
 			continue
 		}
 
-		virtualService := &v1alpha3.VirtualService{}
-		err = json.Unmarshal(jsonBytes, virtualService)
+		meta := &v1.TypeMeta{}
+		err = json.Unmarshal(jsonBytes, meta)
 		if err != nil {
-			log.Debug("error while trying to unmarshal virtualservice", zapcore.Field{Key: "file", String: file})
+			log.Debug("error extracting the metadata of the virtualservice", zapcore.Field{Key: "file", Type: zapcore.StringType, String: file})
 			continue
 		}
 
-		if virtualService.Kind != "VirtualService" {
-			log.Debug("file is not Kind VirtualService", zapcore.Field{Key: "file", String: file})
+		if meta.Kind != "VirtualService" {
+			log.Debug("file is not Kind VirtualService", zapcore.Field{Key: "file", Type: zapcore.StringType, String: file})
 			continue
 		}
 
+		virtualService := &v1alpha3.VirtualService{}
+		err = json.Unmarshal(jsonBytes, virtualService)
+		if err != nil {
+			log.Debug("error while trying to unmarshal virtualservice", zapcore.Field{Key: "file", Type: zapcore.StringType, String: file})
+			return out, fmt.Errorf("error while trying to unmarshal virtualservice (%s): %s", file, err.Error())
```

Comment:
Let's use error wrapping.
```suggestion
			return out, fmt.Errorf("error while trying to unmarshal virtualservice (%s): %w", file, err)
```